h/traversal/contexts.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+"""
+"Context resource" objects for Pyramid views.
```

Comment:
Most excellent. I came into this PR review ready to give the feedback that "we need very comprehensive commenting here" — you've nailed it so far!